tools/scripts/unit-arm.js

```diff
@@ -53,16 +53,19 @@ args.push('-t');
 args.push('5000000');
 
 files.forEach(function (file) {
-  if (file.length > 0 && file.trim()[0] !== '#') {
-    // trim trailing \r if it exists
-    file = file.replace('\r', '');
+    if (file.length > 0 && file.trim()[0] !== '#') {
+        // trim trailing \r if it exists
+        const lastIndex = file.lastIndexOf('\r');
+        if (lastIndex == file.length - 1) {
+            file = files.slice(0, lastIndex) + files.slice(lastIndex).replace('\r', '');
+        }
```

Comment:
This can be improved, the trailing \r is the last one of the string, we can use this `file.endsWith('\r') ? file.slice(0, -1) : file;`

Reply:
Done. Thanks!